### DISCUSSION TITLE
Jetson nano vs (Raspberry Pi 5 8GB + Hailo)

### DISCUSSION THREAD

**Discussion by MECHAAMER** [ultralytics/ultralytics#17741] [Docs]
I'm reaching out to inquire about the best hardware solution for my object detection project. I'm aiming for a minimum of 10 FPS, high accuracy, and stable results. for that I may need yolov5su or yolov5m6u.

I've read that the Jetson Nano offers good performance and power efficiency, while the Raspberry Pi with Hailo accelerator seems to have a more complex setup and less community support.

I'm a bit confused about which option would be the most suitable for my needs. I would appreciate your expert advice on this matter.

Thank you for your time and assistance.

Best regards,

**Comment by UltralyticsAssistant** [comment]
👋 Hello @MECHAAMER, thank you for your interest in Ultralytics and your insightful question about hardware options for object detection projects 🚀! 

For newcomers, I recommend visiting our [Docs](https://docs.ultralytics.com) where you'll find both [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. This might offer some valuable insights that can inform your hardware decision.

If this is a 🐛 Bug Report, kindly provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) so we can better assist you.

For detailed technical queries such as your current one regarding custom hardware solutions, please include any specifics about the models you intend to deploy and the kind of datasets you're working with. This information can be crucial in providing precise advice.

Join our lively community where you can connect with others who may have experience with Jetson Nano or Raspberry Pi setups. For real-time assistance, check out our [Discord](https://ultralytics.com/discord) 🎧. Prefer in-depth discussions? Visit our [Discourse](https://community.ultralytics.com). Or join conversations on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure you are running the latest version of `ultralytics` along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a compatible [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

Consider testing your project in some of our verified environments that come with preinstalled dependencies:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

Check the current status of the repository to ensure all features are functioning as intended by verifying our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule). If the badge is green, all tests are passing:

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

Rest assured, an Ultralytics engineer will also assist you soon. Happy experimenting! 😊

**Reply by MECHAAMER** [reply]
I update my post, I mentioned the Yolo versions: 5su or 5m6u

**Reply by glenn-jocher** [reply]
@MECHAAMER thank you for the update. Both YOLOv5su and YOLOv5m6u are excellent options for edge devices, with YOLOv5m6u offering higher accuracy at the cost of more resources. Your hardware choice should balance model requirements with your device's capacity for optimal performance.

**Reply by MECHAAMER** [reply]
Thank you for your reply,

Can you help me to choice which the best, is it Jetson Nano or R.PI5 8GB +Hailo.

Which one has computing power enough to meet my requirements?

Which one supported more by ultralytics?

**Reply by glenn-jocher** [reply]
Thank you for your question. Based on your requirements, Jetson Nano may not meet the compute demands for YOLOv5m6u. The Raspberry Pi 5 with Hailo offers more computational power but has a more complex setup. Regarding Ultralytics support, both are compatible, but Jetson devices, including advanced models, are better documented and optimized within our ecosystem. Consider your project's complexity and resource requirements for your choice.

**Comment by glenn-jocher** [comment]
@MECHAAMER based on your requirements for running YOLOv5 models at 10+ FPS, the Jetson Nano would be the more straightforward choice. The integrated NVIDIA GPU and extensive optimization tools like TensorRT make it better suited for deep learning inference compared to the Raspberry Pi 5 + Hailo combination, which requires additional setup and configuration. The Jetson Nano also has strong community support and documentation specifically for YOLO deployments. For detailed performance metrics on YOLO models with the IMX500/Raspberry Pi setup, you can reference the benchmarks in our documentation at https://docs.ultralytics.com/integrations/sony-imx500/.
